### PR TITLE
Detail limitations of CODEOWNERS with respect to Enterprise Managed Users

### LINKED ISSUE #16910: Limitations of CODEOWNERS files with respect to Enterprise Managed Users are not detailed
### Code of Conduct

- [X] I have read and agree to the GitHub Docs project's [Code of Conduct](https://github.com/github/docs/blob/main/CODE_OF_CONDUCT.md)

### What article on docs.github.com is affected?

https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/customizing-your-repository/about-code-owners

### What part(s) of the article would you like to see updated?

From my understanding, when used with [Enterprise Managed User](https://docs.github.com/en/enterprise-cloud@latest/admin/identity-and-access-management/managing-iam-with-enterprise-managed-users/about-enterprise-managed-users) identities, CODEOWNERS files do not support the use of email addresses.

The current documentation does not currently indicate as such.

I would propose that the documentation be revised to indicate that email addresses cannot be used in the context of Enterprise Managed Users being in place.

### Additional information

N/A

### PR DESCRIPTION
**Pull request by ReidWeb**
### Why:

Closes #16910

### What's being changed:

I have added a reusable component, and conditionally embedded it following the syntax used for other items.

While this differs from the proposed resolution detailed in #16910 I believe having the note inline with the documentation regarding syntax is a better solution than having in the "About Code Owners" section. Though I am happy to change this if you wish.

### Check off the following:

- [x] I have reviewed my changes in staging (look for "Automatically generated comment" and click **Modified** to view your latest changes).
- [X] For content changes, I have completed the [self-review checklist](https://github.com/github/docs/blob/main/contributing/self-review.md#self-review).

### Writer impact (This section is for GitHub staff members only):

- [ ] This pull request impacts the contribution experience
  - [ ] I have added the 'writer impact' label
  - [ ] I have added a description and/or a video demo of the changes below (e.g. a "before and after video")

### CODE REVIEW

**Comment by ReidWeb** [comment]
I have reviewed my [content alteration in the preview environment](https://docs-17057-4a861f.preview.ghdocs.com/en/repositories/managing-your-repositorys-settings-and-features/customizing-your-repository/about-code-owners) however due to the inclusion of conditionals, I cannot test this change.

Any recommendations for testing this out, would be welcome if you require me to do so before this can be merged.

**Comment by ramyaparimi** [comment]
@ReidWeb 
Thanks so much for opening a PR! I'll get this triaged for review :zap:

**Review by lecoursen — COMMENTED** [review]
Thanks for opening this PR so quickly! Feel free to re-request me for a review after you address my feedback. 😊

File: content/repositories/managing-your-repositorys-settings-and-features/customizing-your-repository/about-code-owners.md
Comment (ReidWeb) [reply content/repositories/managing-your-repositorys-settings-and-features/customizing-your-repository/about-code-owners.md]: Thank you Laura, changes pushed to address this feedback.

**Comment by ReidWeb** [comment]
[Deployment appears to have failed](https://github.com/github/docs/actions/runs/2168564917)

File: content/repositories/managing-your-repositorys-settings-and-features/customizing-your-repository/about-code-owners.md
Comment (lecoursen) [inline content/repositories/managing-your-repositorys-settings-and-features/customizing-your-repository/about-code-owners.md:52]: Whoops, missed this one in my original suggestion!

```suggestion
{% ifversion fpt or ghec%}In most cases, you{% else %}You{% endif %} can also refer to a user by an email address that has been added to their account on {% ifversion ghae %}{% data variables.product.product_name %}{% else %}{% data variables.product.product_location %}{% endif %}, for example `user@example.com`. {% ifversion fpt or ghec %} You cannot use an email address to refer to a {% data variables.product.prodname_managed_user %}. For more information about {% data variables.product.prodname_managed_users %}, see "[About {% data variables.product.prodname_emus %}](/enterprise-cloud@latest/admin/identity-and-access-management/managing-iam-with-enterprise-managed-users/about-enterprise-managed-users){% ifversion fpt %}" in the {% data variables.product.prodname_ghe_cloud %} documentation.{% else %}."{% endif %}{% endif %}
```

**Review by lecoursen — APPROVED** [review]

**Pull request merged**
